frontend/src/components/pages/Form.jsx

```diff
@@ -0,0 +1,200 @@
+import React from "react";
+import Input from "../Input"
+import Button from "../Button/Button";
+
+
+export default function Form() {
+  const [formData, setForm] = React.useState({
+      firstName: '',
+      lastName: '',
+      birthDate: '',
+      sex: false,
+      hasError: false,
+    })
+
+  const [isDataSent, setIsDataSent] = React.useState(false)
+
+  React.useEffect(() => {
+    let timer;
+    if (isDataSent) {
+      timer = setTimeout(() => {
+        setIsDataSent(false);
+      }, 5000);
+    }
+    return () => clearTimeout(timer);
+  }, [isDataSent])
+
+  const handleChange = (event) => {
+    const { name, value, type, checked } = event.target
+    setForm((prevFormData) => ({
+        ...prevFormData,
+        [name]: type === 'radio' ? (checked ? value : prevFormData[name]) : value,
+    }))
+  }
+
+  const formatDate = (dateString) => {
+    const [day, month, year] = dateString.split('.')
+    return `${day.padStart(2, "0")}.${month.padStart(2, "0")}.${year}`
+  }
+
+  const isValidDate = (dateString) => {
+    const regex = /^\d{1,2}\.\d{1,2}\.\d{4}$/
+    if (!regex.test(dateString)) return false
+  
+    const [day, month, year] = dateString.split('.').map(Number)
+    const date = new Date(year, month - 1, day)
+  
+    return (
+      date.getFullYear() === year &&
+      date.getMonth() === month - 1 &&
+      date.getDate() === day
+    )
+  }
+
+  const validateForm = () => {
+    const { firstName, lastName, birthDate, sex } = formData
+    return (
+      firstName.trim() !== '' &&
+      lastName.trim() !== '' &&
+      isValidDate(birthDate) &&
+      sex !== false
+    )
+  }
+  const isFormValid = validateForm()
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    if (validateForm()) {
+      const formattedDate = {
+        ...formData,
+        birthDate: formatDate(formData.birthDate),
+      }
+      console.log('Form Data Submitted:', formattedDate)
+
+      setForm((prevFormData) => ({
+        ...prevFormData,
+        hasError: false,
+      }))
+      sendData(formattedDate)
+    } else {
+      setForm((prevFormData) => ({
+        ...prevFormData,
+        hasError: true,
+      }));
+    }
+  }
+
+  const sendData = (dataToSend) => {
+    fetch("http://localhost:5000/save-client", {
```

Comment:
místo fetche je fajn použít třeba axios :) je to jednodužší

Reply:
jasny, přepsal jsem funkci za použití axios :)